desktop/apps/artwork/components/auction/helpers.coffee

```diff
@@ -9,6 +10,33 @@ pluralize = (word, count, irregular = null) ->
 
 module.exports = {
   liveAuctionUrl,
+  getBidderStatus: (me, auction) ->
+
+    # Due to falsy behavior we need to check for the existence of a
+    # qualified_for_bidding key and then check its value. This handles conditions
+    # where we want to display the "Bid Now" button, but only to trigger a
+    # login modal or redirect to register transition for the user.
+    foundKey = get(me, 'bidders.0', {}).hasOwnProperty('qualified_for_bidding')
+
+    if foundKey
+      isQualified = get(me, 'bidders.0.qualified_for_bidding')
+      if isQualified
+        'qualified'
+      else
+        'registration-pending'
+    else if me && auction.require_bidder_approval
+      'register-for-bid-approval'
+    else if me
+      'bidding-enabled'
```

Comment:
(apologies for the nitpick) This naming is a little confusing to me, just because I would expect someone who is `'qualified'` to also have `'bidding-enabled'`, but I see that in the template code these cases get different treatments. Maybe this status can be more specific: `'logged-in'` or `'not-registered'` or `'can-bid-and-register'`?